src/main/kotlin/org/elm/workspace/ElmWorkspaceService.kt

```diff
@@ -115,7 +115,10 @@ class ElmWorkspaceService(
 
 
     fun useToolchain(toolchain: ElmToolchain?) {
-        modifySettings { it.copy(binDirPath = toolchain?.binDirPath.toString()) }
+        modifySettings {
+            it.copy(binDirPath = toolchain?.binDirPath.toString(),
+                    activateElmFormatOnSaveHook = toolchain?.activateElmFormatOnSaveHook ?: false)
```

Comment:
hmm. I think the case for passing a `null` toolchain here is just for the integration tests so that we can restore to a blank slate between test runs. This means that the default value of `false` for `activateElmFormatOnSaveHook` is being specified in 2 separate places. We should either factor it out, or have a a "default/missing" toolchain value which we use when the `ElmToolchain` parameter is null.

Reply:
I'm reused the constant value here now. Not sure what you mean with the toolchain. Could you please elaborate?